zyte_api/__main__.py

```diff
@@ -14,45 +14,63 @@
     create_session,
     AsyncClient,
 )
+from zyte_api.aio.retry import RetryFactory, _is_throttling_error
 from zyte_api.constants import ENV_VARIABLE, API_URL
 from zyte_api.utils import _guess_intype
-from zyte_api.aio.retry import RetryFactory, _is_throttling_error
 
 
 class DontRetryErrorsFactory(RetryFactory):
     retry_condition = retry_if_exception(_is_throttling_error)
 
 
-logger = logging.getLogger('zyte_api')
+logger = logging.getLogger("zyte_api")
 
 _UNSET = object()
 
 
-async def run(queries, out, *, n_conn, stop_on_errors, api_url,
-              api_key=None, retry_errors=True):
+async def run(
+    queries,
+    out,
+    *,
+    n_conn,
+    stop_on_errors,
+    api_url,
+    api_key=None,
+    retry_errors=True,
+    store_errors=None,
+):
+    def write_output(content):
+        json.dump(content, out, ensure_ascii=False)
+        out.write("\n")
+        out.flush()
+        pbar.update()
+
     retrying = None if retry_errors else DontRetryErrorsFactory().build()
-    client = AsyncClient(n_conn=n_conn, api_key=api_key, api_url=api_url,
-                         retrying=retrying)
+    client = AsyncClient(
+        n_conn=n_conn, api_key=api_key, api_url=api_url, retrying=retrying
+    )
     async with create_session(connection_pool_size=n_conn) as session:
         result_iter = client.request_parallel_as_completed(
             queries=queries,
             session=session,
         )
-        pbar = tqdm.tqdm(smoothing=0, leave=True, total=len(queries), miniters=1,
-                         unit="url")
+        pbar = tqdm.tqdm(
+            smoothing=0, leave=True, total=len(queries), miniters=1, unit="url"
+        )
         pbar.set_postfix_str(str(client.agg_stats))
         try:
             for fut in result_iter:
                 try:
                     result = await fut
-                    json.dump(result, out, ensure_ascii=False)
-                    out.write("\n")
-                    out.flush()
-                    pbar.update()
                 except Exception as e:
+                    logger.error(str(e))
```

Comment:
I believe this line was OK in its old position, because `raise` below would effectively log the error to stdout or stderr, only the write_output call needs to happen earlier than the `raise`.